StaySphere.Api/StaySphere.Infrastructure/Persistence/Repositories/RepositoryBase.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.EntityFrameworkCore;
+using StaySphere.Domain.Common;
+using StaySphere.Domain.Exeptions;
+using StaySphere.Domain.Interfaces.Repositories;
+
+namespace StaySphere.Infrastructure.Persistence.Repositories
+{
+    public class RepositoryBase<T> : IRepositoryBase<T> where T : EntityBase
+    {
+        protected readonly StaySphereDbContext _context;
+        public RepositoryBase(StaySphereDbContext context)
+        {
+            _context = context;
+        }
+        public async Task<T> Create(T entity)
+        {
+            var createEntity = await _context.Set<T>().AddAsync(entity);
+
+            return createEntity.Entity;
+        }
+
+        public async Task Delete(int id)
```

Comment:
Asynchronous  method names should end with "Async" suffix. [You can read here](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/task-asynchronous-programming-model)